src/test/java/org/opensearch/neuralsearch/query/NeuralSparseQueryIT.java

```diff
@@ -206,6 +282,55 @@ public void testBooleanQuery_withMultipleSparseEncodingQueries() {
         }
     }
 
+    /**
+     * Tests bool should query with query text when two phase enabled:
+     * {
+     *     "query": {
+     *         "bool" : {
+     *             "should": [
+     *                "neural_sparse": {
+     *                  "field1": {
+     *                      "query_text": "Hello world a b",
+     *                      "model_id": "dcsdcasd"
+     *                    }
+     *                 },
+     *                "match": {
+     *                  "field2": "Hello world a b",
+     *                 }
+     *             ]
+     *         }
+     *     }
+     * }
+     */
+    @SneakyThrows
+    public void testBooleanQuery_withMultipleSparseEncodingQueries_whenTwoPhaseEnabled() {
+        String modelId = null;
+        try {
+            initializeIndexIfNotExist(TEST_MULTI_NEURAL_SPARSE_FIELD_INDEX_NAME);
+            modelId = prepareSparseEncodingModel();
```

Comment:
We need to rework this integ test. There are 12 new tests where we're uploading a model. Recently we have found that in 2.13+ for jdk21 that causes exception from ml-commons CB when tests are running in infra CI. For now the only working solution is to minimize number of model deployments/undeployments. 

Please consider moving most of test cases to unit test. Integ tests should be used to test end to end flow, not to test all permutations and negative cases.

Reply:
+1 on this comment.

Reply:
I replaced the model inference with a fixed query token in most of the newly added test cases.

Reply:
After we changed to use query_tokens, we don't deploy new model for new neural sparse integ test.